// @stardust-ui/react/dist/commonjs/components/Toolbar/ToolbarMenuItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _invoke2 = _interopRequireDefault(require("lodash/invoke"));

var React = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var PropTypes = _interopRequireWildcard(require("prop-types"));

var customPropTypes = _interopRequireWildcard(require("@stardust-ui/react-proptypes"));

var _lib = require("../../lib");

var _types = require("../../types");

var _accessibility = require("../../lib/accessibility");

var _Box = _interopRequireDefault(require("../Box/Box"));

var _Icon = _interopRequireDefault(require("../Icon/Icon"));

var ToolbarMenuItem =
/*#__PURE__*/
function (_UIComponent) {
  (0, _inherits2.default)(ToolbarMenuItem, _UIComponent);

  function ToolbarMenuItem() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, ToolbarMenuItem);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(ToolbarMenuItem)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "actionHandlers", {
      performClick: function performClick(event) {
        event.preventDefault();

        _this.handleClick(event);
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleClick", function (e) {
      var disabled = _this.props.disabled;

      if (disabled) {
        e.preventDefault();
        return;
      }

      (0, _invoke2.default)(_this.props, 'onClick', e, _this.props);
    });
    return _this;
  }

  (0, _createClass2.default)(ToolbarMenuItem, [{
    key: "renderComponent",
    value: function renderComponent(_ref) {
      var ElementType = _ref.ElementType,
          classes = _ref.classes,
          accessibility = _ref.accessibility,
          unhandledProps = _ref.unhandledProps;
      var _this$props = this.props,
          children = _this$props.children,
          content = _this$props.content,
          disabled = _this$props.disabled,
          icon = _this$props.icon,
          wrapper = _this$props.wrapper;
      var menuItemInner = (0, _lib.childrenExist)(children) ? children : React.createElement(ElementType, (0, _extends2.default)({}, accessibility.attributes.root, unhandledProps, (0, _lib.applyAccessibilityKeyHandlers)(accessibility.keyHandlers.root, unhandledProps), {
        disabled: disabled,
        className: classes.root,
        onClick: this.handleClick
      }), (0, _lib.childrenExist)(children) ? children : React.createElement(React.Fragment, null, _Icon.default.create(icon, {
        defaultProps: {
          xSpacing: !!content ? 'after' : 'none'
        }
      }), content));

      if (!wrapper) {
        return menuItemInner;
      }

      return _Box.default.create(wrapper, {
        defaultProps: Object.assign({
          className: (0, _classnames.default)(ToolbarMenuItem.slotClassNames.wrapper, classes.wrapper)
        }, accessibility.attributes.wrapper),
        overrideProps: function overrideProps() {
          return {
            children: menuItemInner
          };
        }
      });
    }
  }]);
  return ToolbarMenuItem;
}(_lib.UIComponent);

(0, _defineProperty2.default)(ToolbarMenuItem, "displayName", 'ToolbarMenuItem');
(0, _defineProperty2.default)(ToolbarMenuItem, "className", 'ui-toolbar__menuitem');
(0, _defineProperty2.default)(ToolbarMenuItem, "slotClassNames", {
  wrapper: "".concat(ToolbarMenuItem.className, "__wrapper")
});
(0, _defineProperty2.default)(ToolbarMenuItem, "create", void 0);
(0, _defineProperty2.default)(ToolbarMenuItem, "propTypes", Object.assign({}, _lib.commonPropTypes.createCommon(), {
  active: PropTypes.bool,
  disabled: PropTypes.bool,
  icon: customPropTypes.itemShorthand,
  onClick: PropTypes.func,
  wrapper: customPropTypes.itemShorthand
}));
(0, _defineProperty2.default)(ToolbarMenuItem, "defaultProps", {
  as: 'button',
  accessibility: _accessibility.menuItemBehavior,
  wrapper: {
    as: 'li'
  }
});
ToolbarMenuItem.create = (0, _lib.createShorthandFactory)({
  Component: ToolbarMenuItem,
  mappedProp: 'content'
});
/**
 * A ToolbarMenuItem renders ToolbarMenu item as button.
 */

var _default = (0, _types.withSafeTypeForAs)(ToolbarMenuItem);

exports.default = _default;
