// @stardust-ui/react/dist/commonjs/themes/base/components/Input/inputVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _lib = require("../../../../lib");

var _default = function _default(siteVars) {
  return {
    backgroundColor: siteVars.colors.grey[50],
    backgroundColorInverted: siteVars.colors.white,
    borderColor: siteVars.colors.grey[300],
    borderRadius: (0, _lib.pxToRem)(3),
    borderWidth: "1px",
    fontColor: siteVars.colors.grey[600],
    fontSize: (0, _lib.pxToRem)(14),
    iconPosition: 'absolute',
    iconRight: (0, _lib.pxToRem)(10),
    iconColor: siteVars.colors.grey[800],
    iconLeft: (0, _lib.pxToRem)(6),
    inputPaddingWithIconAtStart: "".concat((0, _lib.pxToRem)(7), " ").concat((0, _lib.pxToRem)(12), " ").concat((0, _lib.pxToRem)(7), " ").concat((0, _lib.pxToRem)(24)),
    inputPaddingWithIconAtEnd: "".concat((0, _lib.pxToRem)(7), " ").concat((0, _lib.pxToRem)(24), " ").concat((0, _lib.pxToRem)(7), " ").concat((0, _lib.pxToRem)(12)),
    inputPadding: "".concat((0, _lib.pxToRem)(7), " ").concat((0, _lib.pxToRem)(12)),
    inputFocusBorderColor: siteVars.colors.brand[300],
    inputFocusBorderRadius: (0, _lib.pxToRem)(3),
    placeholderColor: siteVars.colors.grey[600]
  };
};

exports.default = _default;
