// @stardust-ui/react/dist/commonjs/lib/debug/debugDataTraversal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _utils = require("./utils");

var traverseComponentThemeStyles = function traverseComponentThemeStyles(componentStyles, filterData) {
  if (!componentStyles) return {};
  return Object.keys(componentStyles).reduce(function (acc, slotName) {
    var slotStyles = componentStyles[slotName];
    var filteredSlotStyles = filterData(slotStyles);
    return (0, _utils.isNotNullOrEmpty)(filteredSlotStyles) ? Object.assign({}, acc, (0, _defineProperty2.default)({}, slotName, filteredSlotStyles)) : acc;
  }, {});
};

var traverseComponentStyles = function traverseComponentStyles(stylesDebugOutput, filterData) {
  var filteredThemes = stylesDebugOutput.themes.map(function (theme) {
    return traverseComponentThemeStyles(theme.resolved, filterData);
  });
  var filteredInstance = filterData(stylesDebugOutput.instanceOverrides && stylesDebugOutput.instanceOverrides.resolved || {});
  var filteredResult = traverseComponentThemeStyles(stylesDebugOutput.result, filterData);
  return Object.assign({}, (0, _utils.isNotEmptyObjectsArray)(filteredThemes) && {
    themes: filteredThemes
  }, (0, _utils.isNotNullOrEmpty)(filteredInstance) && {
    instanceOverrides: filteredInstance
  }, (0, _utils.isNotNullOrEmpty)(filteredResult) && {
    result: filteredResult
  });
};

var traverseComponentVariables = function traverseComponentVariables(variablesDebugOutput, filterData) {
  var filteredThemes = variablesDebugOutput.themes.map(function (theme) {
    return filterData(theme.resolved);
  });
  var filteredInstance = filterData(variablesDebugOutput.instanceOverrides && variablesDebugOutput.instanceOverrides.resolved || {});
  var filteredResult = filterData(variablesDebugOutput.result);
  return Object.assign({}, (0, _utils.isNotEmptyObjectsArray)(filteredThemes) && {
    themes: filteredThemes
  }, (0, _utils.isNotNullOrEmpty)(filteredInstance) && {
    instanceOverrides: filteredInstance
  }, (0, _utils.isNotNullOrEmpty)(filteredResult) && {
    result: filteredResult
  });
};

var traverse = function traverse(debugOutput, filterData) {
  var stylesResult = traverseComponentStyles(debugOutput.styles, filterData);
  var variablesResult = traverseComponentVariables(debugOutput.variables, filterData);
  var siteVariablesResult = traverseComponentVariables(debugOutput.siteVariables, filterData);
  var result = Object.assign({}, (0, _utils.isNotNullOrEmpty)(stylesResult) && {
    styles: stylesResult
  }, (0, _utils.isNotNullOrEmpty)(variablesResult) && {
    variables: variablesResult
  }, (0, _utils.isNotNullOrEmpty)(siteVariablesResult) && {
    siteVariables: siteVariablesResult
  });
  return (0, _utils.isNotNullOrEmpty)(result) ? Object.assign({
    componentName: debugOutput.componentName
  }, result) : {};
};

var _default = traverse;
exports.default = _default;
