// @stardust-ui/react/dist/commonjs/lib/accessibility/Behaviors/Tree/treeBehavior.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var keyboardKey = _interopRequireWildcard(require("keyboard-key"));

var _types = require("../../types");

var _FocusZone = require("../../FocusZone");

/**
 * @specification
 * Adds role 'tree' to 'root' slot.
 * Adds attribute 'aria-labelledby' based on the property 'aria-labelledby' to 'root' slot.
 * Embeds component into FocusZone.
 * Provides arrow key navigation in vertical direction.
 * Triggers 'expandSiblings' action with '*' on 'root'.
 */
var treeBehavior = function treeBehavior(props) {
  return {
    attributes: {
      root: {
        role: 'tree',
        'aria-labelledby': props['aria-labelledby']
      }
    },
    keyActions: {
      root: {
        expandSiblings: {
          keyCombinations: [{
            keyCode: keyboardKey['*']
          }]
        }
      }
    },
    focusZone: {
      mode: _types.FocusZoneMode.Embed,
      props: {
        direction: _FocusZone.FocusZoneDirection.vertical
      }
    }
  };
};

var _default = treeBehavior;
exports.default = _default;
