// @stardust-ui/react/dist/commonjs/themes/base/components/Checkbox/checkboxVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _lib = require("../../../../lib");

var _default = function _default(siteVars) {
  return {
    background: 'transparent',
    borderColor: siteVars.colors.grey[300],
    borderStyle: "solid",
    borderRadius: (0, _lib.pxToRem)(4),
    borderWidth: (0, _lib.pxToRem)(1),
    checkboxColor: 'transparent',
    gap: (0, _lib.pxToRem)(12),
    margin: '0',
    padding: '0',
    toggleBackground: 'transparent',
    toggleBorderColor: siteVars.colors.grey[300],
    toggleBorderStyle: "solid",
    toggleBorderRadius: (0, _lib.pxToRem)(999),
    toggleBorderWidth: (0, _lib.pxToRem)(1),
    toggleIndicatorColor: 'inherit',
    toggleMargin: '0',
    togglePadding: "0 ".concat((0, _lib.pxToRem)(20), " 0 0"),
    checkedBackground: 'transparent',
    checkedBorderColor: siteVars.colors.grey[500],
    checkboxCheckedColor: siteVars.colors.grey[500],
    checkboxToggleCheckedBackground: 'transparent',
    checkboxToggleCheckedBorderColor: siteVars.colors.grey[500],
    checkboxToggleCheckedColor: 'inherit',
    toggleCheckedPadding: "0 0 0 ".concat((0, _lib.pxToRem)(20)),
    disabledColor: siteVars.colors.grey[300],
    disabledBackground: siteVars.colors.grey[200],
    disabledBorderColor: siteVars.colors.grey[100],
    disabledCheckboxColor: siteVars.colors.grey[300],
    disabledToggleBackground: 'transparent',
    disabledToggleBorderColor: siteVars.colors.grey[200],
    disabledToggleIndicatorColor: siteVars.colors.grey[300]
  };
};

exports.default = _default;
