// @stardust-ui/react/dist/es/lib/debug/utils.js
import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
import _typeof from "@babel/runtime/helpers/esm/typeof";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
export var containsSubstring = function containsSubstring(arg, substring) {
  return typeof arg === 'string' && arg.indexOf(substring) >= 0;
};
export var getLastOf = function getLastOf(array, fallback) {
  return array.length > 0 ? array[array.length - 1] : fallback;
};
export var isNotNullOrEmpty = function isNotNullOrEmpty(object) {
  if (object == null) {
    return false;
  }

  return Object.keys(object).length > 0;
};
export var isNotEmptyObjectsArray = function isNotEmptyObjectsArray(array) {
  if (array == null) {
    return false;
  }

  return array.some(isNotNullOrEmpty);
};
export var deepPickBy = function deepPickBy(object, predicate) {
  var seenObjects = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : [];

  if (!object) {
    return {};
  }

  if (seenObjects.some(function (seenObject) {
    return seenObject === object;
  })) {
    throw new Error("Circular dependency detected.");
  }

  return Object.keys(object).reduce(function (acc, currentPropName) {
    if (predicate(currentPropName, object[currentPropName])) {
      return Object.assign({}, acc, _defineProperty({}, currentPropName, object[currentPropName]));
    }

    if (_typeof(object[currentPropName]) === 'object') {
      var deepPickResult = deepPickBy(object[currentPropName], predicate, [].concat(_toConsumableArray(seenObjects), [object]));

      if (isNotNullOrEmpty(deepPickResult)) {
        return Object.assign({}, acc, _defineProperty({}, currentPropName, deepPickResult));
      }
    }

    return acc;
  }, {});
};
export var deepPick = function deepPick(object, propName) {
  return deepPickBy(object, function (currentPropName) {
    return propName === currentPropName;
  });
};
