// @stardust-ui/react/dist/commonjs/themes/base/components/Dialog/dialogStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _default = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables;
    return {
      background: v.rootBackground,
      borderRadius: v.rootBorderRadius,
      outline: 'none',
      padding: v.rootPadding,
      position: 'relative',
      width: v.rootWidth,
      display: 'grid',
      gridTemplateColumns: '1fr auto'
    };
  },
  actions: function actions() {
    return {
      gridRow: 3,
      gridColumn: '1 / span 2',
      '-ms-grid-column-align': 'end',
      justifySelf: 'right'
    };
  },
  content: function content(_ref2) {
    var v = _ref2.variables;
    return {
      margin: v.contentMargin,
      gridColumn: '1 / span 2',
      gridRow: 2,
      justifySelf: 'left'
    };
  },
  header: function header(_ref3) {
    var v = _ref3.variables;
    return {
      margin: v.headerMargin,
      gridRow: 1,
      gridColumn: 1,
      justifySelf: 'left'
    };
  },
  headerAction: function headerAction() {
    return {
      gridRow: 1,
      gridColumn: 2
    };
  },
  overlay: function overlay(_ref4) {
    var p = _ref4.props,
        v = _ref4.variables;
    return Object.assign({
      alignItems: 'center',
      background: 'transparent',
      bottom: 0,
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      left: 0,
      overflow: 'auto',
      position: 'fixed',
      right: 0,
      top: 0,
      zIndex: v.overlayZIndex
    }, p.backdrop && {
      background: v.overlayBackground
    });
  }
};
exports.default = _default;
