// @stardust-ui/react/dist/es/themes/teams/components/Toolbar/toolbarMenuItemStyles.js
import { getColorScheme } from '../../colors';
var toolbarMenuItemStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables;
    var colors = getColorScheme(v.colorScheme);
    return Object.assign({
      color: v.menuItemForeground || colors.foreground1,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      display: 'flex',
      alignItems: 'center',
      width: '100%',
      maxWidth: '100%',
      padding: v.menuItemPadding,
      cursor: 'pointer',
      ':focus': {
        outline: 0
      },
      ':hover': {
        color: v.menuItemForegroundHover || colors.menuItemForegroundHover,
        backgroundColor: v.menuItemBackgroundHover || colors.menuItemBackgroundHover
      },
      ':focus-visible': {
        color: v.menuItemForegroundFocus || colors.menuItemForegroundFocus,
        backgroundColor: v.menuItemBackgroundFocus || colors.menuItemBackgroundFocus
      }
    }, p.disabled && {
      cursor: 'default',
      color: v.menuItemForegroundDisabled || colors.foregroundDisabled1,
      backgroundColor: v.menuItemBackgroundDisabled,
      ':hover': {// empty to overwrite all existing hover styles
      }
    });
  },
  wrapper: {
    display: 'block'
  }
};
export default toolbarMenuItemStyles;
