// @stardust-ui/react/dist/es/themes/teams/components/Toolbar/toolbarItemStyles.js
import getIconFillOrOutlineStyles from '../../getIconFillOrOutlineStyles';
import { getColorScheme } from '../../colors';
var toolbarItemStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables;
    var colors = getColorScheme(v.colorScheme);
    return Object.assign({
      backgroundColor: v.background,
      borderWidth: v.borderWidth,
      borderStyle: 'solid',
      borderColor: 'transparent',
      borderRadius: v.borderRadius,
      height: v.itemHeight,
      minWidth: v.itemHeight,
      padding: v.itemPadding,
      color: v.foreground || colors.foreground1,
      cursor: 'pointer',
      ':focus': {
        outline: 0
      }
    }, p.active && Object.assign({
      color: v.foregroundActive || colors.foregroundActive,
      backgroundColor: v.backgroundActive
    }, getIconFillOrOutlineStyles({
      outline: false
    })), {
      ':hover': Object.assign({
        color: v.foregroundHover || colors.foregroundHover,
        backgroundColor: v.backgroundHover || colors.backgroundHover
      }, getIconFillOrOutlineStyles({
        outline: false
      })),
      ':focus-visible': Object.assign({
        color: v.foregroundFocus || colors.foregroundFocus,
        backgroundColor: v.backgroundFocus || colors.backgroundFocus,
        borderColor: v.borderFocus || colors.borderFocus
      }, getIconFillOrOutlineStyles({
        outline: false
      }))
    }, p.disabled && {
      color: v.foregroundDisabled || colors.foregroundDisabled1,
      backgroundColor: v.backgroundDisabled,
      cursor: 'default',
      ':hover': {// empty to overwrite all existing hover styles
      }
    });
  }
};
export default toolbarItemStyles;
