// @stardust-ui/react/dist/es/themes/teams/components/Toolbar/toolbarCustomItemStyles.js
import { getColorScheme } from '../../colors';
var toolbarCustomItemStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables;
    var colors = getColorScheme(v.colorScheme);
    return Object.assign({
      backgroundColor: v.background,
      borderColor: 'transparent',
      borderWidth: v.borderWidth,
      borderStyle: 'solid',
      height: v.itemHeight,
      color: v.foreground || colors.foreground1,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center'
    }, p.fitted !== true && p.fitted !== 'horizontally' && {
      paddingLeft: v.customItemHorizontalPadding,
      paddingRight: v.customItemHorizontalPadding
    }, p.fitted !== true && p.fitted !== 'vertically' && {
      paddingTop: v.customItemVerticalPadding,
      paddingBottom: v.customItemVerticalPadding
    }, {
      ':focus': {
        outline: 0
      },
      ':focus-visible': {
        color: v.foregroundFocus || colors.foregroundFocus,
        backgroundColor: v.backgroundFocus || colors.backgroundFocus,
        borderColor: v.borderFocus || colors.borderFocus
      }
    });
  }
};
export default toolbarCustomItemStyles;
