// @stardust-ui/react/dist/commonjs/themes/teams/components/Toolbar/toolbarItemStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _getIconFillOrOutlineStyles = _interopRequireDefault(require("../../getIconFillOrOutlineStyles"));

var _colors = require("../../colors");

var toolbarItemStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables;
    var colors = (0, _colors.getColorScheme)(v.colorScheme);
    return Object.assign({
      backgroundColor: v.background,
      borderWidth: v.borderWidth,
      borderStyle: 'solid',
      borderColor: 'transparent',
      borderRadius: v.borderRadius,
      height: v.itemHeight,
      minWidth: v.itemHeight,
      padding: v.itemPadding,
      color: v.foreground || colors.foreground1,
      cursor: 'pointer',
      ':focus': {
        outline: 0
      }
    }, p.active && Object.assign({
      color: v.foregroundActive || colors.foregroundActive,
      backgroundColor: v.backgroundActive
    }, (0, _getIconFillOrOutlineStyles.default)({
      outline: false
    })), {
      ':hover': Object.assign({
        color: v.foregroundHover || colors.foregroundHover,
        backgroundColor: v.backgroundHover || colors.backgroundHover
      }, (0, _getIconFillOrOutlineStyles.default)({
        outline: false
      })),
      ':focus-visible': Object.assign({
        color: v.foregroundFocus || colors.foregroundFocus,
        backgroundColor: v.backgroundFocus || colors.backgroundFocus,
        borderColor: v.borderFocus || colors.borderFocus
      }, (0, _getIconFillOrOutlineStyles.default)({
        outline: false
      }))
    }, p.disabled && {
      color: v.foregroundDisabled || colors.foregroundDisabled1,
      backgroundColor: v.backgroundDisabled,
      cursor: 'default',
      ':hover': {// empty to overwrite all existing hover styles
      }
    });
  }
};
var _default = toolbarItemStyles;
exports.default = _default;
