// @stardust-ui/react/dist/es/themes/base/components/Loader/loaderStyles.js
import { pxToRem } from '../../../../lib';
var rootFlexDirections = {
  above: 'column-reverse',
  below: 'column',
  start: 'row-reverse',
  end: 'row'
};
export default {
  root: function root(_ref) {
    var p = _ref.props;
    return {
      alignItems: 'center',
      display: p.inline ? 'inline-flex' : 'flex',
      justifyContent: 'center',
      flexDirection: rootFlexDirections[p.labelPosition]
    };
  },
  indicator: function indicator(_ref2) {
    var p = _ref2.props,
        t = _ref2.theme,
        v = _ref2.variables;
    var animationName = {
      keyframe: function keyframe(_ref3) {
        var from = _ref3.from,
            to = _ref3.to;
        return {
          from: {
            transform: "rotate(".concat(from, ")")
          },
          to: {
            transform: "rotate(".concat(to, ")")
          }
        };
      },
      params: {
        from: '0deg',
        to: '360deg'
      }
    };
    var borderColor = "".concat(v.foregroundColor, " ").concat(v.backgroundColor, " ").concat(v.backgroundColor);
    return {
      animationName: animationName,
      animationDuration: '1.3s',
      animationIterationCount: 'infinite',
      animationTimingFunction: 'cubic-bezier(0.53, 0.21, 0.29, 0.67)',
      borderColor: borderColor,
      borderRadius: '50%',
      borderStyle: 'solid',
      borderWidth: v.borderSizes[p.size],
      boxSizing: 'border-box',
      width: v.indicatorSizes[p.size],
      height: v.indicatorSizes[p.size]
    };
  },
  label: {
    margin: pxToRem(10)
  }
};
