// @stardust-ui/react/dist/commonjs/themes/teams/components/List/listItemStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _lib = require("../../../../lib");

var _accessibilityStyles = require("../../../../lib/accessibility/Styles/accessibilityStyles");

var _ListItem = _interopRequireDefault(require("../../../../components/List/ListItem"));

var truncateStyle = {
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
};

var selectableHoverStyle = function selectableHoverStyle(p, v) {
  var _ref;

  return _ref = {
    background: v.selectableFocusHoverBackgroundColor,
    color: v.selectableFocusHoverColor,
    cursor: 'pointer'
  }, (0, _defineProperty2.default)(_ref, "& .".concat(_ListItem.default.slotClassNames.header), {
    color: 'inherit'
  }), (0, _defineProperty2.default)(_ref, "& .".concat(_ListItem.default.slotClassNames.content), {
    color: 'inherit'
  }), (0, _defineProperty2.default)(_ref, "& .".concat(_ListItem.default.slotClassNames.headerMedia), Object.assign({}, _accessibilityStyles.screenReaderContainerStyles, {
    color: 'inherit'
  })), (0, _defineProperty2.default)(_ref, "& .".concat(_ListItem.default.slotClassNames.contentMedia), {
    display: 'none',
    color: 'inherit'
  }), (0, _defineProperty2.default)(_ref, "& .".concat(_ListItem.default.slotClassNames.endMedia), {
    display: 'block',
    color: 'inherit'
  }), _ref;
};

var selectableFocusStyle = function selectableFocusStyle(p, v) {
  return Object.assign({}, selectableHoverStyle(p, v), {
    outline: 0,
    ':focus-visible': {
      outline: ".2rem solid ".concat(v.selectedFocusOutlineColor),
      zIndex: 1
    }
  });
};

var selectedStyle = function selectedStyle(variables) {
  return {
    background: variables.selectedBackgroundColor,
    color: variables.selectedColor
  };
};

var listItemStyles = {
  root: function root(_ref2) {
    var _Object$assign;

    var p = _ref2.props,
        v = _ref2.variables;
    return Object.assign({
      minHeight: v.minHeight,
      padding: v.rootPadding
    }, (p.selectable || p.navigable) && Object.assign((_Object$assign = {
      position: 'relative'
    }, (0, _defineProperty2.default)(_Object$assign, "& .".concat(_ListItem.default.slotClassNames.endMedia), {
      display: 'none'
    }), (0, _defineProperty2.default)(_Object$assign, '&:hover', selectableHoverStyle(p, v)), (0, _defineProperty2.default)(_Object$assign, '&:focus', selectableFocusStyle(p, v)), _Object$assign), p.selected && selectedStyle(v)), p.important && {
      fontWeight: 'bold'
    });
  },
  media: function media(_ref3) {
    var p = _ref3.props;
    return Object.assign({}, p.important && {
      '::before': {
        content: '""',
        position: 'absolute',
        left: (0, _lib.pxToRem)(8),
        width: (0, _lib.pxToRem)(2),
        height: (0, _lib.pxToRem)(2),
        background: '#000'
      }
    }, (p.header || p.content) && {
      marginRight: (0, _lib.pxToRem)(8)
    });
  },
  header: function header(_ref4) {
    var p = _ref4.props,
        v = _ref4.variables;
    return Object.assign({
      fontSize: v.headerFontSize,
      lineHeight: v.headerLineHeight
    }, p.truncateHeader && truncateStyle, (!p.content || p.headerMedia) && {
      marginRight: (0, _lib.pxToRem)(8)
    });
  },
  headerMedia: function headerMedia(_ref5) {
    var v = _ref5.variables;
    return {
      fontSize: v.headerMediaFontSize,
      lineHeight: v.headerMediaLineHeight,
      alignSelf: 'flex-end'
    };
  },
  content: function content(_ref6) {
    var p = _ref6.props,
        v = _ref6.variables;
    return Object.assign({
      fontSize: v.contentFontSize,
      lineHeight: v.contentLineHeight
    }, p.truncateContent && truncateStyle, (!p.header || p.contentMedia) && {
      marginRight: (0, _lib.pxToRem)(8)
    });
  },
  contentMedia: function contentMedia(_ref7) {
    var p = _ref7.props,
        v = _ref7.variables;
    return {
      fontSize: v.contentMediaFontSize,
      lineHeight: v.contentMediaLineHeight
    };
  },
  endMedia: function endMedia(_ref8) {
    var p = _ref8.props;
    return Object.assign({}, (p.selectable || p.navigable) && {
      display: 'none'
    }, {
      flexShrink: 0
    });
  },
  main: function main() {
    return {
      minWidth: 0 // needed for the truncate styles to work

    };
  }
};
var _default = listItemStyles;
exports.default = _default;
