// @stardust-ui/react/dist/es/themes/teams/components/Slider/sliderStyles.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import Slider from '../../../../components/Slider/Slider';
import getBorderFocusStyles from '../../getBorderFocusStyles'; // this selector is used to identify the thumb slot from a previous sibling

var thumbFromPreviousSiblingSelector = "&+ .".concat(Slider.slotClassNames.thumb);
var sliderStyles = {
  input: function input(_ref) {
    var v = _ref.variables,
        siteVariables = _ref.theme.siteVariables;
    var activeThumbStyles = {
      height: v.activeThumbHeight,
      width: v.activeThumbWidth,
      background: v.activeThumbColor,
      marginTop: "calc(".concat(v.height, " / 2  - ").concat(v.activeThumbHeight, " / 2)"),
      marginLeft: "calc(-".concat(v.activeThumbWidth, " / 2)")
    };
    var borderFocusStyles = getBorderFocusStyles({
      siteVariables: siteVariables,
      borderPadding: v.thumbBorderPadding
    });
    return {
      ':active': _defineProperty({}, thumbFromPreviousSiblingSelector, activeThumbStyles),
      ':focus': _defineProperty({}, thumbFromPreviousSiblingSelector, borderFocusStyles[':focus']),
      ':focus-visible': _defineProperty({}, thumbFromPreviousSiblingSelector, Object.assign({}, borderFocusStyles[':focus-visible'], activeThumbStyles))
    };
  }
};
export default sliderStyles;
