// @stardust-ui/react/dist/commonjs/themes/teams/components/Checkbox/checkboxStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _Checkbox = _interopRequireDefault(require("../../../../components/Checkbox/Checkbox"));

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

var checkboxStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables,
        t = _ref.theme;
    return Object.assign({
      color: v.textColor,
      padding: v.rootPadding,
      verticalAlign: 'middle',
      alignItems: 'start'
    }, (0, _getBorderFocusStyles.default)({
      siteVariables: t.siteVariables,
      borderRadius: '3px'
    }), {
      ':hover': (0, _defineProperty2.default)({
        color: v.textColorHover
      }, "& .".concat(_Checkbox.default.slotClassNames.indicator), Object.assign({}, p.checked && {
        background: v.checkedBackgroundHover
      }, !p.checked && Object.assign({
        borderColor: v.borderColorHover
      }, p.toggle && {
        color: v.borderColorHover
      })))
    }, p.checked && {
      color: v.checkedTextColor
    }, p.disabled && {
      pointerEvents: 'none',
      color: v.disabledColor
    });
  },
  checkbox: function checkbox(_ref2) {
    var p = _ref2.props,
        v = _ref2.variables;
    return Object.assign({
      background: v.background,
      borderColor: v.borderColor,
      borderStyle: v.borderStyle,
      borderRadius: v.borderRadius,
      borderWidth: v.borderWidth,
      color: v.indicatorColor,
      margin: v.margin,
      padding: v.padding
    }, p.checked && {
      background: v.checkedBackground,
      borderColor: v.checkedBorderColor,
      color: v.checkedIndicatorColor
    }, p.disabled && {
      background: v.disabledBackground,
      borderColor: v.disabledBorderColor
    }, p.disabled && p.checked && {
      color: v.disabledCheckedIndicatorColor,
      background: v.disabledBackgroundChecked,
      borderColor: 'transparent'
    });
  },
  toggle: function toggle(_ref3) {
    var p = _ref3.props,
        v = _ref3.variables;
    return Object.assign((0, _defineProperty2.default)({
      background: v.background,
      borderColor: v.borderColor,
      borderStyle: v.borderStyle,
      borderRadius: v.toggleBorderRadius,
      borderWidth: v.borderWidth,
      color: v.borderColor,
      margin: v.toggleMargin,
      padding: v.togglePadding,
      transition: 'padding .3s ease'
    }, "& svg", {
      width: v.toggleIndicatorSize,
      height: v.toggleIndicatorSize
    }), p.checked && {
      background: v.checkedBackground,
      borderColor: v.checkedBorderColor,
      color: v.checkedIndicatorColor,
      padding: v.toggleCheckedPadding
    }, p.disabled && {
      color: v.disabledToggleIndicatorColor,
      background: v.disabledBackground,
      borderColor: v.disabledBorderColor
    }, p.disabled && p.checked && {
      color: v.disabledCheckedIndicatorColor,
      background: v.disabledBackgroundChecked,
      borderColor: 'transparent'
    });
  }
};
var _default = checkboxStyles;
exports.default = _default;
