// @stardust-ui/react/dist/commonjs/themes/base/componentVariables.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Checkbox", {
  enumerable: true,
  get: function get() {
    return _checkboxVariables.default;
  }
});
Object.defineProperty(exports, "Dialog", {
  enumerable: true,
  get: function get() {
    return _dialogVariables.default;
  }
});
Object.defineProperty(exports, "Flex", {
  enumerable: true,
  get: function get() {
    return _flexVariables.default;
  }
});
Object.defineProperty(exports, "FlexItem", {
  enumerable: true,
  get: function get() {
    return _flexItemVariables.default;
  }
});
Object.defineProperty(exports, "Icon", {
  enumerable: true,
  get: function get() {
    return _iconVariables.default;
  }
});
Object.defineProperty(exports, "Input", {
  enumerable: true,
  get: function get() {
    return _inputVariables.default;
  }
});
Object.defineProperty(exports, "Loader", {
  enumerable: true,
  get: function get() {
    return _loaderVariables.default;
  }
});
Object.defineProperty(exports, "ProviderBox", {
  enumerable: true,
  get: function get() {
    return _providerBoxVariables.default;
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _textVariables.default;
  }
});
Object.defineProperty(exports, "Grid", {
  enumerable: true,
  get: function get() {
    return _gridVariables.default;
  }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function get() {
    return _imageVariables.default;
  }
});
Object.defineProperty(exports, "Slider", {
  enumerable: true,
  get: function get() {
    return _sliderVariables.default;
  }
});

var _checkboxVariables = _interopRequireDefault(require("./components/Checkbox/checkboxVariables"));

var _dialogVariables = _interopRequireDefault(require("./components/Dialog/dialogVariables"));

var _flexVariables = _interopRequireDefault(require("./components/Flex/flexVariables"));

var _flexItemVariables = _interopRequireDefault(require("./components/Flex/flexItemVariables"));

var _iconVariables = _interopRequireDefault(require("./components/Icon/iconVariables"));

var _inputVariables = _interopRequireDefault(require("./components/Input/inputVariables"));

var _loaderVariables = _interopRequireDefault(require("./components/Loader/loaderVariables"));

var _providerBoxVariables = _interopRequireDefault(require("./components/Provider/providerBoxVariables"));

var _textVariables = _interopRequireDefault(require("./components/Text/textVariables"));

var _gridVariables = _interopRequireDefault(require("./components/Grid/gridVariables"));

var _imageVariables = _interopRequireDefault(require("./components/Image/imageVariables"));

var _sliderVariables = _interopRequireDefault(require("./components/Slider/sliderVariables"));
