// @stardust-ui/react/dist/commonjs/themes/base/siteVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "colors", {
  enumerable: true,
  get: function get() {
    return _colors.colors;
  }
});
Object.defineProperty(exports, "contextualColors", {
  enumerable: true,
  get: function get() {
    return _colors.contextualColors;
  }
});
Object.defineProperty(exports, "emphasisColors", {
  enumerable: true,
  get: function get() {
    return _colors.emphasisColors;
  }
});
Object.defineProperty(exports, "naturalColors", {
  enumerable: true,
  get: function get() {
    return _colors.naturalColors;
  }
});
Object.defineProperty(exports, "colorScheme", {
  enumerable: true,
  get: function get() {
    return _colors.colorScheme;
  }
});
exports.lineHeightLargest = exports.lineHeightLarger = exports.lineHeightLarge = exports.lineHeightMedium = exports.lineHeightSmall = exports.lineHeightSmaller = exports.lineHeightSmallest = exports.fontWeightBold = exports.fontWeightSemibold = exports.fontWeightRegular = exports.fontWeightSemilight = exports.fontWeightLight = exports.fontSizes = void 0;

var _colors = require("./colors");

//
// COLORS
//
//
// FONT SIZES
//
var fontSizes = {
  smallest: '0.8rem',
  smaller: '1rem',
  small: '1.2rem',
  medium: '1.4rem',
  large: '1.8rem',
  larger: '2.4rem',
  largest: '3rem' //
  // FONT WEIGHTS
  //

};
exports.fontSizes = fontSizes;
var fontWeightLight = 200;
exports.fontWeightLight = fontWeightLight;
var fontWeightSemilight = 300;
exports.fontWeightSemilight = fontWeightSemilight;
var fontWeightRegular = 400;
exports.fontWeightRegular = fontWeightRegular;
var fontWeightSemibold = 600;
exports.fontWeightSemibold = fontWeightSemibold;
var fontWeightBold = 700; //
// LINE HEIGHTS
//

exports.fontWeightBold = fontWeightBold;
var lineHeightSmallest = 1.2;
exports.lineHeightSmallest = lineHeightSmallest;
var lineHeightSmaller = 1.2;
exports.lineHeightSmaller = lineHeightSmaller;
var lineHeightSmall = 1.3333;
exports.lineHeightSmall = lineHeightSmall;
var lineHeightMedium = 1.4286;
exports.lineHeightMedium = lineHeightMedium;
var lineHeightLarge = 1.3333;
exports.lineHeightLarge = lineHeightLarge;
var lineHeightLarger = 1.3333;
exports.lineHeightLarger = lineHeightLarger;
var lineHeightLargest = 1.3333;
exports.lineHeightLargest = lineHeightLargest;
