// @stardust-ui/react/dist/commonjs/themes/teams/components/Slider/sliderStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _Slider = _interopRequireDefault(require("../../../../components/Slider/Slider"));

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

// this selector is used to identify the thumb slot from a previous sibling
var thumbFromPreviousSiblingSelector = "&+ .".concat(_Slider.default.slotClassNames.thumb);
var sliderStyles = {
  input: function input(_ref) {
    var v = _ref.variables,
        siteVariables = _ref.theme.siteVariables;
    var activeThumbStyles = {
      height: v.activeThumbHeight,
      width: v.activeThumbWidth,
      background: v.activeThumbColor,
      marginTop: "calc(".concat(v.height, " / 2  - ").concat(v.activeThumbHeight, " / 2)"),
      marginLeft: "calc(-".concat(v.activeThumbWidth, " / 2)")
    };
    var borderFocusStyles = (0, _getBorderFocusStyles.default)({
      siteVariables: siteVariables,
      borderPadding: v.thumbBorderPadding
    });
    return {
      ':active': (0, _defineProperty2.default)({}, thumbFromPreviousSiblingSelector, activeThumbStyles),
      ':focus': (0, _defineProperty2.default)({}, thumbFromPreviousSiblingSelector, borderFocusStyles[':focus']),
      ':focus-visible': (0, _defineProperty2.default)({}, thumbFromPreviousSiblingSelector, Object.assign({}, borderFocusStyles[':focus-visible'], activeThumbStyles))
    };
  }
};
var _default = sliderStyles;
exports.default = _default;
