// @stardust-ui/react/dist/commonjs/themes/teams/components/Text/textVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.textColorAreas = void 0;

var _lib = require("../../../../lib");

var _colorUtils = require("../../../colorUtils");

var textColorAreas = (0, _lib.stringLiteralsArray)('foreground');
exports.textColorAreas = textColorAreas;

var _default = function _default(siteVariables) {
  return {
    colorScheme: (0, _colorUtils.pickValuesFromColorScheme)(siteVariables.colorScheme, textColorAreas),
    atMentionOtherColor: siteVariables.colors.brand[600],
    atMentionMeColor: siteVariables.colors.orange[400],
    atMentionMeFontWeight: siteVariables.fontWeightBold,
    disabledColor: siteVariables.colors.grey[250],
    errorColor: siteVariables.colors.red[400],
    importantWeight: siteVariables.fontWeightBold,
    importantColor: siteVariables.colors.red[400],
    successColor: siteVariables.colors.green[600],
    timestampColor: siteVariables.colors.grey[350],
    timestampHoverColor: siteVariables.colors.grey[500]
  };
};

exports.default = _default;
