// @stardust-ui/react/dist/es/themes/teams/components/Button/buttonVariables.js
import { pxToRem } from '../../../../lib';
export default (function (siteVars) {
  return {
    padding: "0 ".concat(pxToRem(20)),
    height: pxToRem(32),
    minWidth: pxToRem(96),
    loadingMinWidth: pxToRem(118),
    maxWidth: pxToRem(280),
    borderRadius: siteVars.borderRadius,
    contentFontSize: siteVars.fontSizes.medium,
    contentFontWeight: siteVars.fontWeightSemibold,
    contentLineHeight: siteVars.lineHeightMedium,
    color: siteVars.colors.grey[750],
    colorHover: siteVars.colors.grey[750],
    colorFocus: siteVars.colors.grey[750],
    colorDisabled: siteVars.colorScheme.brand.foregroundDisabled,
    backgroundColor: siteVars.colors.white,
    backgroundColorActive: siteVars.colors.grey[200],
    backgroundColorHover: siteVars.colors.grey[50],
    backgroundColorFocus: siteVars.colors.grey[200],
    backgroundColorDisabled: siteVars.colors.grey[150],
    borderColor: siteVars.colors.grey[200],
    borderColorHover: siteVars.colors.grey[250],
    borderColorDisabled: 'transparent',
    primaryColor: siteVars.colors.white,
    primaryColorHover: siteVars.colors.white,
    primaryColorFocus: siteVars.colors.white,
    primaryBackgroundColor: siteVars.colors.brand[600],
    primaryBackgroundColorActive: siteVars.colors.brand[900],
    primaryBackgroundColorHover: siteVars.colors.brand[800],
    primaryBackgroundColorFocus: siteVars.colors.brand[800],
    primaryBorderColor: 'transparent',
    circularBorderRadius: pxToRem(999),
    circularColor: siteVars.colors.grey[500],
    circularColorActive: siteVars.colors.white,
    circularBackgroundColor: 'transparent',
    circularBackgroundColorActive: siteVars.colors.grey[500],
    circularBackgroundColorHover: siteVars.colors.grey[450],
    circularBackgroundColorFocus: siteVars.colors.grey[450],
    circularBorderColor: siteVars.colors.grey[500],
    circularBorderColorHover: 'transparent',
    circularBorderColorFocus: 'transparent',
    textColor: siteVars.colorScheme.default.foreground1,
    textColorHover: siteVars.colorScheme.brand.foreground1,
    textPrimaryColor: siteVars.colorScheme.brand.foreground,
    textPrimaryColorHover: siteVars.colorScheme.brand.foreground1,
    textColorDisabled: siteVars.colorScheme.brand.foregroundDisabled1,
    boxShadow: siteVars.shadowLevel1,
    loaderBorderSize: pxToRem(2),
    loaderSize: pxToRem(20),
    loaderSvgHeight: pxToRem(1220),
    loaderSvgAnimationHeight: pxToRem(-1200),
    sizeSmallContentFontSize: siteVars.fontSizes.small,
    sizeSmallContentLineHeight: siteVars.lineHeightSmall,
    sizeSmallHeight: pxToRem(24),
    sizeSmallMinWidth: pxToRem(72),
    sizeSmallPadding: "0 ".concat(pxToRem(8)),
    sizeSmallLoaderBorderSize: pxToRem(2),
    sizeSmallLoaderSize: pxToRem(15),
    sizeSmallLoaderSvgHeight: pxToRem(895),
    sizeSmallLoaderSvgAnimationHeight: pxToRem(-880)
  };
});
