// @stardust-ui/react/dist/commonjs/themes/teams/components/Button/buttonVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _lib = require("../../../../lib");

var _default = function _default(siteVars) {
  return {
    padding: "0 ".concat((0, _lib.pxToRem)(20)),
    height: (0, _lib.pxToRem)(32),
    minWidth: (0, _lib.pxToRem)(96),
    loadingMinWidth: (0, _lib.pxToRem)(118),
    maxWidth: (0, _lib.pxToRem)(280),
    borderRadius: siteVars.borderRadius,
    contentFontSize: siteVars.fontSizes.medium,
    contentFontWeight: siteVars.fontWeightSemibold,
    contentLineHeight: siteVars.lineHeightMedium,
    color: siteVars.colors.grey[750],
    colorHover: siteVars.colors.grey[750],
    colorFocus: siteVars.colors.grey[750],
    colorDisabled: siteVars.colorScheme.brand.foregroundDisabled,
    backgroundColor: siteVars.colors.white,
    backgroundColorActive: siteVars.colors.grey[200],
    backgroundColorHover: siteVars.colors.grey[50],
    backgroundColorFocus: siteVars.colors.grey[200],
    backgroundColorDisabled: siteVars.colors.grey[150],
    borderColor: siteVars.colors.grey[200],
    borderColorHover: siteVars.colors.grey[250],
    borderColorDisabled: 'transparent',
    primaryColor: siteVars.colors.white,
    primaryColorHover: siteVars.colors.white,
    primaryColorFocus: siteVars.colors.white,
    primaryBackgroundColor: siteVars.colors.brand[600],
    primaryBackgroundColorActive: siteVars.colors.brand[900],
    primaryBackgroundColorHover: siteVars.colors.brand[800],
    primaryBackgroundColorFocus: siteVars.colors.brand[800],
    primaryBorderColor: 'transparent',
    circularBorderRadius: (0, _lib.pxToRem)(999),
    circularColor: siteVars.colors.grey[500],
    circularColorActive: siteVars.colors.white,
    circularBackgroundColor: 'transparent',
    circularBackgroundColorActive: siteVars.colors.grey[500],
    circularBackgroundColorHover: siteVars.colors.grey[450],
    circularBackgroundColorFocus: siteVars.colors.grey[450],
    circularBorderColor: siteVars.colors.grey[500],
    circularBorderColorHover: 'transparent',
    circularBorderColorFocus: 'transparent',
    textColor: siteVars.colorScheme.default.foreground1,
    textColorHover: siteVars.colorScheme.brand.foreground1,
    textPrimaryColor: siteVars.colorScheme.brand.foreground,
    textPrimaryColorHover: siteVars.colorScheme.brand.foreground1,
    textColorDisabled: siteVars.colorScheme.brand.foregroundDisabled1,
    boxShadow: siteVars.shadowLevel1,
    loaderBorderSize: (0, _lib.pxToRem)(2),
    loaderSize: (0, _lib.pxToRem)(20),
    loaderSvgHeight: (0, _lib.pxToRem)(1220),
    loaderSvgAnimationHeight: (0, _lib.pxToRem)(-1200),
    sizeSmallContentFontSize: siteVars.fontSizes.small,
    sizeSmallContentLineHeight: siteVars.lineHeightSmall,
    sizeSmallHeight: (0, _lib.pxToRem)(24),
    sizeSmallMinWidth: (0, _lib.pxToRem)(72),
    sizeSmallPadding: "0 ".concat((0, _lib.pxToRem)(8)),
    sizeSmallLoaderBorderSize: (0, _lib.pxToRem)(2),
    sizeSmallLoaderSize: (0, _lib.pxToRem)(15),
    sizeSmallLoaderSvgHeight: (0, _lib.pxToRem)(895),
    sizeSmallLoaderSvgAnimationHeight: (0, _lib.pxToRem)(-880)
  };
};

exports.default = _default;
